### PR TITLE
Refactor/member

### PR DESCRIPTION
**Pull request by m0uadh880**

### CODE REVIEW

File: src/main/database/reset_mysql_password.sql [hunk 1]
```diff
@@ -0,0 +1,2 @@
+ALTER USER 'root'@'localhost' IDENTIFIED BY 'password';
+FLUSH PRIVILEGES;
```
Comment (ounitech) [inline src/main/database/reset_mysql_password.sql:2]: as discussed not needed

File: src/main/java/com/ounitech/wemove/controllers/MemberController.java [hunk 2]
```diff
@@ -174,4 +175,71 @@ public ResponseEntity<Page<Member>> findMembersWithPagination(@PathVariable int
...
+            Optional<MemberSubscription> optionalMemberSubscription = memberSubscriptionService.findMemberSubscription(member);
+
+            if (optionalMemberSubscription.isPresent()) {
+                memberSubscriptionService.deleteMemberSubscription(optionalMemberSubscription.get().getId());
+                return new ResponseEntity<>(HttpStatus.OK);
+            }
+        }
+
+        return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+    }
```
Comment (ounitech) [inline src/main/java/com/ounitech/wemove/controllers/MemberController.java:244]: thanks :+1:

File: src/main/resources/application.properties [hunk 3]
```diff
@@ -1 +1,2 @@
-spring.application.name=wemove
\ No newline at end of file
+spring.application.name=wemove
+spring.profiles.active=prod
```
Comment (ounitech) [inline src/main/resources/application.properties:2]: please delete this line.
you can dig further about spring profiles in spring boot documentation 
https://docs.spring.io/spring-boot/reference/index.html

File: src/test/java/com/ounitech/wemove/repositories/EntryRepositoryTest.java [hunk 4]
```diff
@@ -6,13 +6,15 @@
...
 import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.test.context.TestPropertySource;
 
 import java.util.List;
 import java.util.function.Function;
 
 import static org.assertj.core.api.Assertions.assertThat;
 
 @DataJpaTest
+@TestPropertySource(locations = "classpath:application-test.properties")
```
Comment (ounitech) [inline src/test/java/com/ounitech/wemove/repositories/EntryRepositoryTest.java:17]: not needed as discussed

File: src/test/java/com/ounitech/wemove/repositories/MemberSubscriptionRepositoryTest.java [hunk 5]
```diff
@@ -1,15 +1,45 @@
...
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.test.context.TestPropertySource;
 
-import static org.junit.jupiter.api.Assertions.*;
+import java.util.Optional;
 
+import static org.assertj.core.api.Assertions.assertThat;
+
+@DataJpaTest
+@TestPropertySource(locations = "classpath:application-test.properties")
```
Comment (ounitech) [inline src/test/java/com/ounitech/wemove/repositories/MemberSubscriptionRepositoryTest.java:18]: as discussed, no need

File: src/test/java/com/ounitech/wemove/repositories/StaffRepositoryTest.java [hunk 6]
```diff
@@ -7,12 +7,14 @@
...
 import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
 import org.springframework.boot.test.autoconfigure.orm.jpa.TestEntityManager;
+import org.springframework.test.context.TestPropertySource;
 
 import java.util.List;
 
 import static org.assertj.core.api.Assertions.assertThat;
 
 @DataJpaTest
+@TestPropertySource(locations = "classpath:application-test.properties")
```
Comment (ounitech) [inline src/test/java/com/ounitech/wemove/repositories/StaffRepositoryTest.java:17]: as discussed, not needed

**Review by ounitech — CHANGES_REQUESTED** [review]

File: src/test/java/com/ounitech/wemove/repositories/EntryRepositoryTest.java (hunk 4 above)
Comment (m0uadh880) [reply src/test/java/com/ounitech/wemove/repositories/EntryRepositoryTest.java:17]: The @DataJpaTest annotation sets up a default embedded database, so there's no need to add a test environment to the application. My question is: is it better to override the default setup by adding a test environment, or we can rely on the @DataJpaTest configuration and remove the application-test.properties file?

Comment (ounitech) [reply src/test/java/com/ounitech/wemove/repositories/EntryRepositoryTest.java]: you can remove application-test.properties. @DataJpaTest defaults suits our needs for now

**Pull request merged**